Explorer/Assets/DCL/RealmNavigation/TeleportController.cs

```diff
@@ -58,9 +59,16 @@ public void InvalidateRealm()
             retrieveScene = null;
         }
 
+        public void SetTerrain(TerrainGenerator terrain)
+        {
+            this.terrain = terrain;
+        }
+
         private async UniTask<WaitForSceneReadiness?> TeleportAsync(Vector2Int parcel, PickTargetDelegate pickTargetDelegate,
             AsyncLoadProcessReport loadReport, CancellationToken ct)
         {
+            terrain.SetTerrainCollider(parcel, true);
```

Comment:
Do you really need this? Looks like the coupling that starts and then is difficult to extract.

 When you teleport, wont the system that turns on the parcel take effect. I mean, since you come from Mordor, it should recognize it

Reply:
As discussed in PM, I will address this and other comments in the upcoming branch.

Reply:
Solved [here](https://github.com/decentraland/unity-explorer/pull/3826)